model/src/main/java/com/chiralbehaviors/CoRE/Ruleform.java

```diff
@@ -90,6 +109,36 @@
         CONCRETE_SUBCLASSES = Collections.unmodifiableMap(concrete);
     }
 
+    public static <T extends Ruleform> T deserialize(InputStream is,
```

Comment:
moved all the Util methods over to Ruleforms to allow accessing of private variables (i.e. "id") to avoid premature (and fatal) proxy unwrapping in hibernate.